internal/generator/vector/output/splunk/splunk_sink_with_indexed_fields.toml

```diff
@@ -0,0 +1,64 @@
+# Ensure timestamp field well formatted for Splunk
+[transforms.splunk_hec_timestamp]
+type = "remap"
+inputs = ["pipelineName"]
+source = '''
+ts, err = parse_timestamp(.@timestamp,"%+")
+if err != null {
+	log("could not parse timestamp. err=" + err, rate_limit_secs: 0)
+} else {
+	.@timestamp = ts
+}
+
+'''
+
+[transforms.splunk_hec_metadata]
+type = "remap"
+inputs = ["splunk_hec_timestamp"]
+source = '''
+# Splunk 'source' field detection
+if ._internal.log_type == "infrastructure" && ._internal.log_source == "node" {
+    ._internal.splunk.source = to_string!(._internal.systemd.u.SYSLOG_IDENTIFIER || "")
+}
+if ._internal.log_source == "container" {
+   	._internal.splunk.source = join!([._internal.kubernetes.namespace_name, ._internal.kubernetes.pod_name, ._internal.kubernetes.container_name], "_")
+}
+if ._internal.log_type == "audit" {
+   ._internal.splunk.source = ._internal.log_source
+}
+._internal.splunk.sourcetype = "_json"
+# Splunk indexed fields
+indexed_fields = [["log_source"],["kubernetes","namespace_labels","bar/baz0-9.test"],["annotations","authorization.k8s.io/decision"]]
+for_each(indexed_fields) -> |_, field| {
+    value = get!(., field)
+    if !is_null(value) {
+        new_key = replace(join!(field,"_"), r'[\./]', "_")
+        if !is_string(value) {
+          if is_object(value) {
+            value = encode_json(value)
+          } else {
+            value = to_string!(value)
+          }
+        }
+        . = remove!(., field, true)
```

Comment:
looking at our document I don't see where we state we remove the indexed field from the payload